.github/codeql/queries/unique-command-use.ql

```diff
@@ -0,0 +1,93 @@
+/**
+ * @name A VS Code command should not be used in multiple locations
+ * @kind problem
+ * @problem.severity warning
+ * @id vscode-codeql/unique-command-use
+ * @description Using each VS Code command from only one location makes
```

Comment:
So, the goal is to ensure we use different commands for different kinds of interactions even if they have the same handler? Eg- distinguishing between right-click->Run Query and CMD+Shift+P->Run Query?

And this is only around helping to understand which ways our users are interacting with the product?

Reply:
Yes, your understanding is correct. The goal is to distinguish between different ways of performing the same action, so distinguishing between right-click->Run Query and CMD+Shift+P->Run Query would be an example because one is the editor context menu and the other is the command palette. So they should be registered in different places in package.json